src/util.ts

```diff
@@ -62,57 +51,42 @@ export function normalizeUri(uri: string): string {
 	return url.format(parts);
 }
 
-export function path2uri(root: string, file: string): string {
-	let ret = 'file://';
-	if (!strict && process.platform === 'win32') {
-		ret += '/';
-	}
-	let p;
-	if (root) {
-		p = resolve(root, file);
-	} else {
-		p = file;
-	}
-	if (/^[a-z]:[\\\/]/i.test(p)) {
-		p = '/' + p;
+/**
+ * Converts an abolute path to a file:// uri.
+ * @param path an absolute path
+ */
+export function path2uri(path: string): string {
+	// Require a leading slash, on windows prefixed with drive letter
+	if (!/^([a-z]:)?[\\\/]/i.test(path)) {
+		throw new Error(`Expected an absolute uri, got ${path}`);
 	}
-	p = p.split(/[\\\/]/g).map((val, i) => i <= 1 && /^[a-z]:$/i.test(val) ? val : encodeURIComponent(val)).join('/');
-	return normalizeUri(ret + p);
+
+	const [firstSegment, ...remainingSegments] = path.split(/[\\\/]/);
```

Comment:
I am changing this since spreading is slow: https://jsperf.com/spread-vs-shift/1